custom/icds_reports/static/js/directives/program-summary/program-summary.directive.js

```diff
@@ -149,11 +149,14 @@ function ProgramSummaryController($scope, $http, $log, $routeParams, $location,
         } else {
             vm.moveToLocation(data.location, data.locationLevel);
         }
+        dateHelperService.updateSelectedMonth(data['month'], data['year']);
+        storageService.setKey('search', $location.search());
+        $scope.$emit('filtersChange');
```

Comment:
copied this behavior from the web directive's handling of getting data back from the modal: https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js#L120-L125